Add Stepper Input docs (#3586)
jpmorganchase/salt-ds: site/docs/components/stepper-input/examples.mdx

---
title:
  $ref: ./#/title
layout: DetailComponent
sidebar:
  exclude: true
data:
  $ref: ./#/data
---

<LivePreviewControls>
<LivePreview componentName="stepper-input" exampleName="Default" >

## Default

> Stepper input has a default value of 0, an increment value of 1 and a block value of 10, and you can only use it to accept numeric characters. A single click increases or decreases the value in the input by 1.

Reviewer: Can't really test atm but does it actually have a default of 0?

Reviewer 2: No, removed line

</LivePreview>
<LivePreview componentName="stepper-input" exampleName="Variants" >

## Variants

> The component comes in two variants: primary and secondary. By default, the primary variant is selected. To switch to the secondary variant, use `variant="secondary"`. When choosing between the primary and secondary variants, consider the background color on which the component will be placed.

Reviewer: Not sure about the terminology "the primary variant is selected"

Reviewer 3: How about "By default, the primary variant is displayed."

**Refresh adornment**:

Reset the input to it’s default value, using the refresh button.

**Sync adornment**:

Sync to a live value from a source that is constantly changing, for example, when dealing with live stock prices. The component's value doesn't automatically update when the live value changes, however. In this case, the sync button automatically appears when the displayed value and live value don’t match.

**Custom buttons**:

Customize the positioning and appearance of the stepper buttons to meet specific product requirements or restrictions.

</LivePreview>
<LivePreview componentName="stepper-input" exampleName="Disabled" >

## Disabled

You can disable a stepper input, resulting in no action when the user interacts with it.

</LivePreview>
<LivePreview componentName="stepper-input" exampleName="Readonly" >

## Read-only

You can set stepper input to a read-only state. The user can't edit the value text in this state, but they can highlight and copy it.

By default, the marker is an em dash. If your application requires a different marker, you can set it via the `emptyReadOnlyMarker` prop.

> - Use the read-only state when the user needs the value for their flow or current task but cannot edit it, for example, because of user permissions.

Reviewer: Is this a best practice? There are some short lists without headings (this example and the next one)

---
title:
  $ref: ./#/title
layout: DetailComponent
sidebar:
  exclude: true
data:
  $ref: ./#/data
---

<LivePreviewControls>
<LivePreview componentName="stepper-input" exampleName="Default" >

## Default

Stepper input has an increment step value of 1 and a step block value of 10, and you can only use it to accept numeric characters.

</LivePreview>
<LivePreview componentName="stepper-input" exampleName="Variants" >

## Variants

> The component comes in two variants: primary and secondary, with default set to primary. To switch to the secondary variant, use `variant="secondary"`. When choosing between the primary and secondary variants, consider the background color on which the component will be placed.

Reviewer: ```suggestion
The component comes in two variants: primary and secondary. The primary variant is the default. To switch to the secondary variant, use `variant="secondary"`. 

When choosing between the primary and secondary variants, consider the background color on which the component will be placed.
```
